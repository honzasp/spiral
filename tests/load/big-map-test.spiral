(program
  (import std)
  (import std.map)
  (import std.test)

  (test "big map of integers added in order" (lambda (t)
    (let ((limit 100000)
          (m (do ((i 0 (+ i 1))
                  (m (map-new -)
                     (map-set m i (* 42 i))))
                 ((>= i limit) m))))
      (do ((i 0 (+ i 1)))
          ((>= i limit))
        (let ((k (mod (* i 123) 53)))
          (assert-eqv t (map-get m k) (* 42 k))))))))
