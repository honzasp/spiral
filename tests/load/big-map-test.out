test big map of integers added in order ok
